test/cases/enum.ts

```diff
@@ -20,10 +21,158 @@ export var schema = {
   "additionalProperties": false
 }
 
-export var types = `interface Enum {
+export var configurations = [
+  {
+    settings: {
+      useTypescriptEnums: false
+    },
+    types: `interface Enum {
   foo: "a" | "b" | "c";
   bar: number;
   baz: {
     a: number;
   };
 }`
+  }, 
+  {
+    settings: {
+      useTypescriptEnums: true
+    },
+    types: `enum Foo {
+  a,
+  b,
+  c
+}
+enum Bar {
+  One = 1,
+  Two = 2,
+  Three = 3
+}
+enum Baz {
+  [object Object],
+  [object Object],
+  [object Object]
+}
+interface Enum {
+  foo: Foo;
+  bar: Bar;
+  baz: Baz;
+}`
+  },
+    {
+    settings: {
+      useTypescriptEnums: true,
+      addEnumUtils: true
+    },
+    types: `enum Foo {
+  a,
+  b,
+  c
+}
+class FooUtil {
```

Comment:
Let's avoid generating code, and keep all the checks to compile time. If you'd like to generate  runtime check, please write up an informal proposal and comment in https://github.com/bcherny/json-schema-to-typescript/issues/8.